# Preprocessing

In this notebook, I'll be preprocessing all my data to get it ready to be put into a model.

In [30]:
#load in
import warnings
warnings.filterwarnings("ignore")
import pandas as pd
pd.set_option('display.max_columns', 100)
import numpy as np


## Data Loading

Just checking to see if my data has been correctly imported.

In [37]:
#load in dataset
df=pd.read_csv("data1/my_dataset.csv")
#look at sample
df.sample(5)

player_id  last_season_x  most_recent_club_id country_of_citizenship  \
62241     152963           2022                   44                Germany   
44017     538964           2023                  415            Netherlands   
52803      40372           2020                  430                  Spain   
58464     167799           2023                  148                Denmark   
47874     204155           2022                49702                Hungary   

           sub_position  position   foot  height_in_cm  market_value  \
62241    Centre-Forward    Attack  right         187.0        800000   
44017    Centre-Forward    Attack  right         190.0       2500000   
52803  Central Midfield  Midfield  right         175.0       6000000   
58464  Central Midfield  Midfield  right         185.0        500000   
47874       Centre-Back  Defender  right         190.0        800000   

       highest_ever_market_value        date  age  contract_days_left  month  \
62241                 10000000.0  2023-06-22   33                 597      6   
44017                 12000000.0  2022-09-19   22                 962      9   
52803                 23000000.0  2018-03-24   33                -134      3   
58464                 45000000.0  2013-06-30   18                 597      6   
47874                  1000000.0  2022-06-02   26                 962      6   

       year  goals  assists  yellow_cards  red_cards  minutes_played  \
62241  2023      1        0             2          0             786   
44017  2022      3        1             3          0             972   
52803  2018      1        0             5          0            1928   
58464  2013      0        1             0          0              32   
47874  2022      0        0             1          1             364   

       player_club_id league_id  net_transfer_record  national_team_players  
62241              44        L1            6700000.0                      7  
44017             415       FR1             200000.0                      7  
52803              46       IT1           68000000.0                     18  
58464              27        L1           18250000.0                     18  
47874           49702       RU1            1030000.0                      7

Check if datatypes have been 

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 123594 entries, 0 to 123593
Data columns (total 24 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   player_id                  123594 non-null  int64  
 1   last_season_x              123594 non-null  int64  
 2   most_recent_club_id        123594 non-null  int64  
 3   country_of_citizenship     123594 non-null  object 
 4   sub_position               123594 non-null  object 
 5   position                   123594 non-null  object 
 6   foot                       123594 non-null  object 
 7   height_in_cm               123594 non-null  float64
 8   market_value               123594 non-null  int64  
 9   highest_ever_market_value  123594 non-null  float64
 10  date                       123594 non-null  object 
 11  age                        123594 non-null  int64  
 12  contract_days_left         123594 non-null  int64  
 13  month                      12

- The `player_id` collumn seems relatively useless in the case of predicting market value as its just an identifier.
- The `most_recent_club_id` collumn also seems useless as I already have the player_club_id collumn making it useless.
- We can also drop the `highest_ever_market_value` collumn as its way too correlated (see EDA notebook) with the target variable.

We can drop them both.

In [5]:
df=df.drop(["player_id","most_recent_club_id"],axis=1)

In [6]:
df.describe()

last_season_x   height_in_cm  market_value  highest_ever_market_value  \
count  123594.000000  123594.000000  1.235940e+05               1.235940e+05   
mean     2021.424948     182.265814  5.106035e+06               1.030578e+07   
std         2.298380       6.695002  1.061903e+07               1.747416e+07   
min      2012.000000     159.000000  1.000000e+04               2.500000e+04   
25%      2021.000000     178.000000  5.000000e+05               1.000000e+06   
50%      2023.000000     183.000000  1.500000e+06               3.500000e+06   
75%      2023.000000     187.000000  4.500000e+06               1.200000e+07   
max      2023.000000     207.000000  2.000000e+08               2.000000e+08   

                 age  contract_days_left          month           year  \
count  123594.000000       123594.000000  123594.000000  123594.000000   
mean       25.031830          443.834749       6.428346    2018.691587   
std         3.985102          471.763162       3.587796       2.976855   
min        15.000000         -314.000000       1.000000    2012.000000   
25%        22.000000          232.000000       3.000000    2017.000000   
50%        25.000000          232.000000       6.000000    2019.000000   
75%        28.000000          597.000000      10.000000    2021.000000   
max        44.000000         3154.000000      12.000000    2023.000000   

               goals        assists   yellow_cards      red_cards  \
count  123594.000000  123594.000000  123594.000000  123594.000000   
mean        1.973793       1.501165       2.737706       0.068798   
std         3.695028       2.450115       2.855257       0.265803   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         1.000000       1.000000       2.000000       0.000000   
75%         2.000000       2.000000       4.000000       0.000000   
max        59.000000      33.000000      23.000000       3.000000   

       minutes_played  player_club_id  net_transfer_record  \
count   123594.000000   123594.000000         1.235940e+05   
mean      1308.635727     3175.928945         2.138397e+05   
std       1010.055198     7846.777161         3.819746e+07   
min          1.000000        3.000000        -1.951000e+08   
25%        439.000000      294.000000        -1.930000e+06   
50%       1143.000000      873.000000         1.500000e+05   
75%       1990.000000     2457.000000         7.600000e+06   
max       5070.000000    83678.000000         1.020000e+08   

       national_team_players  
count          123594.000000  
mean                6.551669  
std                 5.306323  
min                 0.000000  
25%                 2.000000  
50%                 6.000000  
75%                10.000000  
max                22.000000

Everything looks good, Lets get going with the preprocessing.

First, to avoid data leakage we will do our train test split before any proprocessing.

## Splitting

In [7]:
#Split my X and y
y=df["market_value"]
X=df.drop(["market_value"],axis=1)

In [8]:
#import splitter
from sklearn.model_selection import train_test_split

#split between train and val_test
X_train, X_val_test, y_train, y_val_test = train_test_split(X, y, test_size=0.33, random_state=100)

In [9]:
#split between test and val
X_val, X_test, y_val, y_test = train_test_split(X_val_test, y_val_test, test_size=0.5, random_state=100)

In [10]:
#check shape
X_train.shape

(82807, 21)

In [11]:
#check shape
X_val.shape

(20393, 21)

In [12]:
#check shape
X_test.shape

(20394, 21)

my splitting seems to have been correctly done. All the rows add up to the original size of the dataframe.

## Feature Encoding

Since I've got the month and year collumn i dont need the date collumn so I'll drop that.

In [13]:
X_train=X_train.drop(["date"],axis=1)
X_test=X_test.drop(["date"],axis=1)
X_val=X_val.drop(["date"],axis=1)

I need to encode the following collumns to be processed:

- `country_of_citizenship`
- `sub_position`
- `date`
- `position` 
- `foot` 
- `league_id`
  

We'll first look at the cardinality of all our categorical collumns to decide what type of encoding we should do.

In [14]:
df.nunique()

last_season_x                  12
country_of_citizenship        155
sub_position                   13
position                        4
foot                            3
height_in_cm                   48
market_value                  223
highest_ever_market_value     157
date                         2141
age                            30
contract_days_left             89
month                          12
year                           12
goals                          51
assists                        31
yellow_cards                   23
red_cards                       4
minutes_played               4079
player_club_id                426
league_id                      14
net_transfer_record           272
national_team_players          23
dtype: int64

the `foot`,`position`,`subposition` and `domestic competition id` collumns all have a low enough cardinality so one hot encoding would definitely be suitable.

In [15]:
#for train
encodecols = ["foot", "position", "sub_position", "league_id"]
X_train = pd.get_dummies(X_train, columns=encodecols, prefix=encodecols)

#for val
X_val=pd.get_dummies(X_val,columns=encodecols,prefix=encodecols)

#for test
X_test=pd.get_dummies(X_test,columns=encodecols,prefix=encodecols)

In [16]:
X_train.sample(2)

last_season_x country_of_citizenship  height_in_cm  \
15784            2023              Argentina         165.0   
121771           2015                 Turkey         172.0   

        highest_ever_market_value  age  contract_days_left  month  year  \
15784                   3000000.0   18                1327      6  2022   
121771                   400000.0   22                -134      1  2015   

        goals  assists  yellow_cards  red_cards  minutes_played  \
15784       1        0             2          0             244   
121771      0        0             4          0             803   

        player_club_id  net_transfer_record  national_team_players  foot_both  \
15784              398           10040000.0                     10          0   
121771             589            9430000.0                      4          0   

        foot_left  foot_right  position_Attack  position_Defender  \
15784           1           0                1                  0   
121771          0           1                0                  1   

        position_Goalkeeper  position_Midfield  \
15784                     0                  0   
121771                    0                  0   

        sub_position_Attacking Midfield  sub_position_Central Midfield  \
15784                                 0                              0   
121771                                0                              0   

        sub_position_Centre-Back  sub_position_Centre-Forward  \
15784                          0                            0   
121771                         0                            0   

        sub_position_Defensive Midfield  sub_position_Goalkeeper  \
15784                                 0                        0   
121771                                0                        0   

        sub_position_Left Midfield  sub_position_Left Winger  \
15784                            0                         0   
121771                           0                         0   

        sub_position_Left-Back  sub_position_Right Midfield  \
15784                        0                            0   
121771                       0                            0   

        sub_position_Right Winger  sub_position_Right-Back  \
15784                           1                        0   
121771                          0                        1   

        sub_position_Second Striker  league_id_BE1  league_id_DK1  \
15784                             0              0              0   
121771                            0              0              0   

        league_id_ES1  league_id_FR1  league_id_GB1  league_id_GR1  \
15784               0              0              0              0   
121771              0              0              0              0   

        league_id_IT1  league_id_L1  league_id_NL1  league_id_PO1  \
15784               1             0              0              0   
121771              0             0              0              0   

        league_id_RU1  league_id_SC1  league_id_TR1  league_id_UKR1  
15784               0              0              0               0  
121771              0              0              1               0

In [17]:
X_test.sample(2)

last_season_x country_of_citizenship  height_in_cm  \
68239           2022                  Ghana         174.0   
47701           2023                 Poland         186.0   

       highest_ever_market_value  age  contract_days_left  month  year  goals  \
68239                  3000000.0   25                -134      2  2016      0   
47701                 40000000.0   24                 962     12  2017      3   

       assists  yellow_cards  red_cards  minutes_played  player_club_id  \
68239        0             2          0             450             167   
47701        0             1          0             474            6195   

       net_transfer_record  national_team_players  foot_both  foot_left  \
68239           17450000.0                      6          0          0   
47701          -26500000.0                     16          0          1   

       foot_right  position_Attack  position_Defender  position_Goalkeeper  \
68239           1                0                  1                    0   
47701           0                1                  0                    0   

       position_Midfield  sub_position_Attacking Midfield  \
68239                  0                                0   
47701                  0                                0   

       sub_position_Central Midfield  sub_position_Centre-Back  \
68239                              0                         0   
47701                              0                         0   

       sub_position_Centre-Forward  sub_position_Defensive Midfield  \
68239                            0                                0   
47701                            1                                0   

       sub_position_Goalkeeper  sub_position_Left Midfield  \
68239                        0                           0   
47701                        0                           0   

       sub_position_Left Winger  sub_position_Left-Back  \
68239                         0                       0   
47701                         0                       0   

       sub_position_Right Midfield  sub_position_Right Winger  \
68239                            0                          0   
47701                            0                          0   

       sub_position_Right-Back  sub_position_Second Striker  league_id_BE1  \
68239                        1                            0              0   
47701                        0                            0              0   

       league_id_DK1  league_id_ES1  league_id_FR1  league_id_GB1  \
68239              0              0              0              0   
47701              0              0              0              0   

       league_id_GR1  league_id_IT1  league_id_L1  league_id_NL1  \
68239              0              0             1              0   
47701              0              1             0              0   

       league_id_PO1  league_id_RU1  league_id_SC1  league_id_TR1  \
68239              0              0              0              0   
47701              0              0              0              0   

       league_id_UKR1  
68239               0  
47701               0

In [18]:
X_val.sample(2)

last_season_x country_of_citizenship  height_in_cm  \
94899           2023                Denmark         174.0   
3738            2023                Denmark         192.0   

       highest_ever_market_value  age  contract_days_left  month  year  goals  \
94899                  3500000.0   20                1327     12  2016      0   
3738                  32000000.0   26                 962     11  2022      1   

       assists  yellow_cards  red_cards  minutes_played  player_club_id  \
94899        2             6          0            2177            2778   
3738         2             4          0            3420             873   

       net_transfer_record  national_team_players  foot_both  foot_left  \
94899           17350000.0                      4          0          1   
3738           -38700000.0                      8          0          0   

       foot_right  position_Attack  position_Defender  position_Goalkeeper  \
94899           0                0                  1                    0   
3738            1                0                  1                    0   

       position_Midfield  sub_position_Attacking Midfield  \
94899                  0                                0   
3738                   0                                0   

       sub_position_Central Midfield  sub_position_Centre-Back  \
94899                              0                         0   
3738                               0                         1   

       sub_position_Centre-Forward  sub_position_Defensive Midfield  \
94899                            0                                0   
3738                             0                                0   

       sub_position_Goalkeeper  sub_position_Left Midfield  \
94899                        0                           0   
3738                         0                           0   

       sub_position_Left Winger  sub_position_Left-Back  \
94899                         0                       1   
3738                          0                       0   

       sub_position_Right Midfield  sub_position_Right Winger  \
94899                            0                          0   
3738                             0                          0   

       sub_position_Right-Back  sub_position_Second Striker  league_id_BE1  \
94899                        0                            0              0   
3738                         0                            0              0   

       league_id_DK1  league_id_ES1  league_id_FR1  league_id_GB1  \
94899              1              0              0              0   
3738               0              0              0              1   

       league_id_GR1  league_id_IT1  league_id_L1  league_id_NL1  \
94899              0              0             0              0   
3738               0              0             0              0   

       league_id_PO1  league_id_RU1  league_id_SC1  league_id_TR1  \
94899              0              0              0              0   
3738               0              0              0              0   

       league_id_UKR1  
94899               0  
3738                0

This seems to have worked well.The only collumn i need to deal with now is the `country_of_citizenship` collumn which consists of 155 unique values which on its own is far too many to one hot encode. The likely solution to this would be to get the top N-countries of citizenship in terms of numbers and have one designated collumn named other for every other nation other than them. This seems viable as nationalities in this dataset arent uniformly distributed as seen in our EDA in the previous notebook as the top 10 nationalities accounted for a disproportionate amount of the dataset.

In [19]:
#get the value counts for each nation
country_counts = df["country_of_citizenship"].value_counts()

#get the top 50 and its index
top_countries=country_counts.head(25).index



In [20]:
top_countries

Index(['Spain', 'France', 'Netherlands', 'Brazil', 'Turkey', 'Germany',
       'Portugal', 'Italy', 'Russia', 'England', 'Ukraine', 'Belgium',
       'Denmark', 'Greece', 'Argentina', 'Scotland', 'Serbia', 'Croatia',
       'Senegal', 'Nigeria', 'Sweden', 'Morocco', 'Uruguay', 'Cote d'Ivoire',
       'Poland'],
      dtype='object')

I need to now set everything not in that index to "Other".

In [21]:
#create lambda function setting "Other" if not in index for train,test and val
X_train["country_of_citizenship"]=X_train["country_of_citizenship"].apply(lambda x: x if x in top_countries else 'Other')
X_val["country_of_citizenship"]=X_val["country_of_citizenship"].apply(lambda x: x if x in top_countries else 'Other')
X_test["country_of_citizenship"]=X_test["country_of_citizenship"].apply(lambda x: x if x in top_countries else 'Other')

In [22]:
X_train.head()

last_season_x country_of_citizenship  height_in_cm  \
76575            2022                  Italy         184.0   
115759           2021                  Spain         180.0   
38683            2022                  Other         187.0   
44111            2023                  Spain         183.0   
90042            2022                  Other         181.0   

        highest_ever_market_value  age  contract_days_left  month  year  \
76575                  18000000.0   24                 597      6  2018   
115759                  5000000.0   20                -134      1  2015   
38683                   4500000.0   26                 232      6  2018   
44111                    800000.0   18                 597      2  2020   
90042                   3000000.0   22                1327     10  2017   

        goals  assists  yellow_cards  red_cards  minutes_played  \
76575       0        0             0          0              23   
115759      1        0             0          0             161   
38683       2        0             6          0             810   
44111       0        0             1          0             281   
90042       0        0             4          0            1211   

        player_club_id  net_transfer_record  national_team_players  foot_both  \
76575                5          -45000000.0                     15          0   
115759             150           43500000.0                      7          0   
38683              618           12850000.0                      3          0   
44111             1108           -6300000.0                      3          0   
90042              618           12850000.0                      3          0   

        foot_left  foot_right  position_Attack  position_Defender  \
76575           0           1                0                  1   
115759          0           1                1                  0   
38683           0           1                0                  0   
44111           1           0                0                  1   
90042           0           1                0                  1   

        position_Goalkeeper  position_Midfield  \
76575                     0                  0   
115759                    0                  0   
38683                     0                  1   
44111                     0                  0   
90042                     0                  0   

        sub_position_Attacking Midfield  sub_position_Central Midfield  \
76575                                 0                              0   
115759                                0                              0   
38683                                 0                              1   
44111                                 0                              0   
90042                                 0                              0   

        sub_position_Centre-Back  sub_position_Centre-Forward  \
76575                          0                            0   
115759                         0                            0   
38683                          0                            0   
44111                          0                            0   
90042                          0                            0   

        sub_position_Defensive Midfield  sub_position_Goalkeeper  \
76575                                 0                        0   
115759                                0                        0   
38683                                 0                        0   
44111                                 0                        0   
90042                                 0                        0   

        sub_position_Left Midfield  sub_position_Left Winger  \
76575                            0                         0   
115759                           0                         1   
38683                            0                         0   
44111                            0                         0   
90042        

I can now perform one hot encoding without excessively increasing dimensionality.

In [23]:


#train
X_train = pd.get_dummies(X_train, columns=["country_of_citizenship"], prefix="Country")

#val
X_val = pd.get_dummies(X_val, columns=["country_of_citizenship"], prefix="Country")

#test
X_test = pd.get_dummies(X_test, columns=["country_of_citizenship"], prefix="Country")


In [24]:
X_train.sample(5)

last_season_x  height_in_cm  highest_ever_market_value  age  \
72517           2023         171.0                   800000.0   22   
2746            2022         181.0                  3000000.0   22   
43038           2023         171.0                  6500000.0   26   
71075           2023         185.0                  5000000.0   24   
62977           2023         180.0                  5000000.0   26   

       contract_days_left  month  year  goals  assists  yellow_cards  \
72517                 597      8  2017      0        0             0   
2746                  232      9  2013      0        0             0   
43038                 567      6  2023      1        3             1   
71075                 232      1  2017      1        2             3   
62977                 962     11  2021      0        0             0   

       red_cards  minutes_played  player_club_id  net_transfer_record  \
72517          0              47             897            -435000.0   
2746           0              67             714            5500000.0   
43038          0            1912             371           12500000.0   
71075          0             987           12321           -4350000.0   
62977          0             118             244          -30600000.0   

       national_team_players  foot_both  foot_left  foot_right  \
72517                      0          0          0           1   
2746                       4          0          1           0   
43038                     16          0          1           0   
71075                      5          0          1           0   
62977                     15          0          0           1   

       position_Attack  position_Defender  position_Goalkeeper  \
72517                1                  0                    0   
2746                 0                  1                    0   
43038                0                  1                    0   
71075                0                  1                    0   
62977                0                  0                    0   

       position_Midfield  sub_position_Attacking Midfield  \
72517                  0                                0   
2746                   0                                0   
43038                  0                                0   
71075                  0                                0   
62977                  1                                0   

       sub_position_Central Midfield  sub_position_Centre-Back  \
72517                              0                         0   
2746                               0                         0   
43038                              0                         0   
71075                              0                         0   
62977                              1                         0   

       sub_position_Centre-Forward  sub_position_Defensive Midfield  \
72517                            0                                0   
2746                             0                                0   
43038                            0                                0   
71075                            0                                0   
62977                            0                                0   

       sub_position_Goalkeeper  sub_position_Left Midfield  \
72517                        0                           0   
2746                         0                           0   
43038                        0                           0   
71075                        0                           0   
62977                        0                           0   

       sub_position_Left Winger  sub_position_Left-Back  \
72517                         0                       0   
2746                          0                       1   
43038                         0                       1   
71075                         0                       1   
62977                         0                       0   

 

I've now completed my feature encoding and can move onto the next step of my preprocessing.

## Scaling Data

I'll use the StandardScaler as there will be quite a few outliers as bigger teams in better leagues have a disproportionate amount of funds to spend on players which min-max scaling is very sensitive to.

In [25]:
from sklearn.preprocessing import StandardScaler
#instantiate
scaler=StandardScaler()

#fit and transform
#train
X_train_s=scaler.fit_transform(X_train)

#val
X_val_s=scaler.transform(X_val)


#test
X_test_s=scaler.transform(X_test)




They're now in array form and I want it back in dataframe format.

In [26]:
#train
X_train_s = pd.DataFrame(X_train_s, columns=X_train.columns)

#val
X_val_s = pd.DataFrame(X_val_s, columns=X_val.columns)


#test
X_test_s = pd.DataFrame(X_test_s, columns=X_test.columns)



In [27]:
X_train_s.sample(10)

last_season_x  height_in_cm  highest_ever_market_value       age  \
60256       0.686577     -0.337849                  -0.407135 -0.010027   
28433      -0.618347     -1.385054                  -0.189545  0.240506   
20617      -2.793220     -1.235453                  -0.533108  0.491039   
41021       0.686577      0.709356                   1.413748  0.240506   
66075      -0.183373     -1.085852                  -0.567464  0.240506   
15415       0.686577     -0.936252                  -0.017764  0.240506   
37460       0.686577     -0.337849                   0.268539 -0.010027   
81134      -1.923271     -0.188248                  -0.573190 -0.761625   
41279       0.686577      0.858957                  -0.475847 -0.761625   
74936      -0.618347      0.709356                  -0.304066  1.743703   

       contract_days_left     month      year     goals   assists  \
60256            1.098831  0.996950  0.777335  0.816965  1.015889   
28433           -1.223367 -0.118134  0.105595 -0.532255 -0.612944   
20617           -0.058030  1.554492 -0.902014 -0.532255 -0.612944   
41021           -0.447889 -0.675677  0.441465 -0.262411  0.201472   
66075            0.325471 -0.118134  0.777335 -0.262411 -0.612944   
15415            1.872191 -0.675677  0.441465  0.007433  1.830305   
37460           -0.447889 -0.675677  0.441465 -0.532255 -0.205736   
81134           -0.447889  0.160637 -1.237884 -0.262411 -0.612944   
41279           -0.447889 -0.396905 -0.230274 -0.532255 -0.612944   
74936           -1.223367  1.554492  0.105595 -0.532255 -0.612944   

       yellow_cards  red_cards  minutes_played  player_club_id  \
60256      0.797245  -0.258501       -0.276274       -0.040419   
28433     -0.960436  -0.258501       -1.245178       -0.345651   
20617     -0.960436  -0.258501       -1.189699       -0.208597   
41021      0.445709  -0.258501        2.271805       -0.276996   
66075     -0.960436  -0.258501       -1.017317       -0.397654   
15415     -0.608900  -0.258501        1.290022       -0.268670   
37460     -0.608900  -0.258501       -0.862768       -0.269823   
81134     -0.608900  -0.258501       -1.115396       -0.236136   
41279     -0.960436  -0.258501       -1.281834       -0.323620   
74936      0.797245  -0.258501        0.121987       -0.315934   

       net_transfer_record  national_team_players  foot_both  foot_left  \
60256            -0.017825              -0.295615  -0.203311  -0.577466   
28433            -0.345352               1.403856  -0.203311  -0.577466   
20617             0.006625              -0.862105   4.918564  -0.577466   
41021            -3.960570               2.914496  -0.203311  -0.577466   
66075             0.001395               0.082045  -0.203311  -0.577466   
15415             2.534009              -0.106785  -0.203311  -0.577466   
37460             2.295260              -0.295615  -0.203311  -0.577466   
81134            -0.002528              -1.050935  -0.203311  -0.577466   
41279            -0.002528              -0.673275  -0.203311  -0.577466   
74936            -0.279716               0.459705  -0.203311  -0.577466   

       foot_right  position_Attack  position_Defender  position_Goalkeeper  \
60256    0.638745         1.574668          -0.710504            -0.297908   
28433    0.638745        -0.635055          -0.710504            -0.297908   
20617   -1.565570         1.574668          -0.710504            -0.297908   
41021    0.638745        -0.635055           1.407452            -0.297908   
66075    0.638745        -0.635055          -0.710504            -0.297908   
15415    0.638745         1.574668          -0.710504            -0.297908   
37460    0.638745        -0.635055           1.407452            -0.297908   
81134    0.638745        -0.635055          -0.710504            -0.297908   
41279    0.638745         1.574668          -0.710504            -0.297908   
74936    0.638745        -0.635055           1.407452            -0.297908   

       po

In [28]:
#single out Cristiano Ronaldo using scaled data
X_train_s.loc[
    (X_train_s['position_Attack'] > 1.0) & (X_train_s['position_Attack'] < 2.0) &
    (X_train_s['Country_Portugal'] > 4.0) & (X_train_s['Country_Portugal'] < 5.0) &
    (X_train_s['foot_right'] > -1.6) & (X_train_s['foot_right'] < -1.5)&
    (X_train_s["height_in_cm"]>0.7) & (X_train_s["height_in_cm"]<0.71)
]



last_season_x  height_in_cm  highest_ever_market_value       age  \
1122        0.686577      0.709356                   0.268539 -0.761625   
3336        0.686577      0.709356                   0.268539 -0.010027   
10245       0.686577      0.709356                   0.268539 -0.761625   
14562       0.686577      0.709356                   0.268539 -0.010027   
26416       0.686577      0.709356                   0.268539  1.493170   
27580       0.686577      0.709356                   0.268539  0.491039   
40603       0.686577      0.709356                   0.268539  0.741572   
55824       0.686577      0.709356                   0.268539  0.992104   
58852       0.686577      0.709356                   0.268539  1.242637   
69160       0.686577      0.709356                   0.268539  1.242637   
76847       0.686577      0.709356                   0.268539  0.741572   
80396       0.686577      0.709356                   0.268539  0.240506   
82494       0.686577      0.709356                   0.268539  0.491039   

       contract_days_left     month      year     goals   assists  \
1122             1.098831  0.160637 -1.573754 -0.262411 -0.612944   
3336             1.098831 -0.118134 -0.566144  1.356653  0.608680   
10245            1.098831 -1.233219 -1.237884 -0.532255 -0.612944   
14562            1.098831  1.275721 -0.566144  1.356653  0.608680   
26416            1.098831 -0.118134  1.449075  2.436029  2.237513   
27580            1.098831 -0.118134  0.105595  3.515405  0.608680   
40603            1.098831  0.439408  0.441465  5.134469  3.459137   
55824            1.098831 -0.396905  0.777335  2.705873  2.644721   
58852            1.098831 -0.396905  1.113205  3.245561  2.644721   
69160            1.098831 -0.954448  1.449075  2.436029  2.237513   
76847            1.098831 -1.511990  0.777335  2.705873  2.644721   
80396            1.098831 -0.118134 -0.230274  3.515405  1.423097   
82494            1.098831 -0.675677  0.441465  5.134469  3.459137   

       yellow_cards  red_cards  minutes_played  player_club_id  \
1122       0.445709  -0.258501       -0.669582       -0.092679   
3336       0.094172  -0.258501        0.156662       -0.265468   
10245     -0.608900  -0.258501       -1.047038       -0.092679   
14562      0.094172  -0.258501        0.156662       -0.265468   
26416      0.797245  -0.258501        0.490527       -0.360124   
27580      0.797245  -0.258501        0.956156       -0.265468   
40603      1.500317  -0.258501        1.822028       -0.265468   
55824      2.203390  -0.258501        1.802214       -0.265468   
58852      1.851853  -0.258501        1.121603       -0.360124   
69160      0.797245  -0.258501        0.490527       -0.360124   
76847      2.203390  -0.258501        1.802214       -0.265468   
80396      0.094172  -0.258501        0.811514       -0.265468   
82494      1.500317  -0.258501        1.822028       -0.265468   

       net_transfer_record  national_team_players  foot_both  foot_left  \
1122              0.204056              -0.673275  -0.203311   1.731702   
3336             -0.177732              -0.106785  -0.203311   1.731702   
10245             0.204056              -0.673275  -0.203311   1.731702   
14562            -0.177732              -0.106785  -0.203311   1.731702   
26416             1.814888               0.270875  -0.203311   1.731702   
27580            -0.177732              -0.106785  -0.203311   1.731702   
40603            -0.177732              -0.106785  -0.203311   1.731702   
55824            -0.177732              -0.106785  -0.203311   1.731702   
58852             1.814888               0.270875  -0.203311   1.731702   
69160             1.814888               0.270875  -0.203311   1.731702   
76847            -0.177732              -0.106785  -0.203311   1.731702   
80396            -0.177732              -0.106785  -0.203311   1.731702   
82494            -0.177732              -0.106785  -0.203311   1.731702   

       foot_r

Data looks correctly scaled. We can now move onto our Baseline linear Model.

### Data Export

In [29]:
import pickle
df.to_pickle("data1/df.pkl")
X_train_s.to_pickle('data1/X_train_s.pkl')
X_val_s.to_pickle('data1/X_val_s.pkl')
X_test_s.to_pickle('data1/X_test_s.pkl')
y_train.to_pickle('data1/y_train.pkl')
y_val.to_pickle('data1/y_val.pkl')
y_test.to_pickle('data1/y_test.pkl')